topics/go/algorithms/data/heap/heap.go

```diff
@@ -0,0 +1,140 @@
+// Package heap implements a heap interface, and add ability to store values of type Data in the minimum Heap.
+package heap
+
+import (
+	"container/heap"
+	"errors"
+)
+
+// ErrEmptyHeap returned if heap is empty
+var ErrEmptyHeap = errors.New("heap is empty")
+
+// Data represents what is being stored on the heap.
+type Data struct {
+	Value int
+	Index int
+}
+
+// Heap represents a list of data.
+type Heap struct {
+	size     int
+	capacity int
+	dataHeap *dataHeap
+}
+
+// New returns a new Heap with given capacity in which we will store our data.
+func New(cap int) (*Heap, error) {
+
+	if cap <= 0 {
+		return nil, errors.New("invalid capacity")
```

Comment:
You added an error variable at the top, which is fine but then why not the consistency here? Let's use them or not. If it is really not needed, then let's not.

Reply:
I'll remove the exported error variable from the top. The main purpose of this error was just checking errors in tests, but we just can check that err! = Nil.